### ISSUE TITLE
Select 1 throws exception on distributed merged table

### ISSUE THREAD

**Issue by yiquanzhou** [ClickHouse/ClickHouse#9043]
**Describe the bug or unexpected behaviour**
```
select 1 from distributed_merged_table
```
throws exception
```
Received exception from server (version 19.16.10):
Code: 171. DB::Exception: Received from clickhouse-server:9000. DB::Exception: Cannot convert column `1` because it is non constant in source stream but must be constant in result.
```
on version `19.16.10.44`

**How to reproduce**
* Use Clickhouse version `19.16.10.44`
* Create some tables: 2 local table, one local merge table. 2 distributed tables, one distributed merge table

```
CREATE TABLE local_table_1(
id String
) ENGINE = ReplicatedMergeTree('/clickhouse/tables/{shard}/default/local_table_1', '{replica}')
ORDER BY (id)

CREATE TABLE local_table_2(
id String
) ENGINE = ReplicatedMergeTree('/clickhouse/tables/{shard}/default/local_table_2', '{replica}')
ORDER BY (id)

CREATE TABLE local_table_merged (
id String
) ENGINE = Merge('default', 'local_table_1|local_table_2');

CREATE TABLE distributed_table_1 (
id String
) ENGINE = Distributed('{cluster}', default, local_table_1);

CREATE TABLE distributed_table_2 (
id String
) ENGINE = Distributed('{cluster}', default, local_table_2);

CREATE TABLE distributed_table_merged (
id String
) ENGINE = Merge('default', 'distributed_table_1|distributed_table_2');
```

* Run query `SELECT 1` on all these tables

**Expected behavior**
Query should return 0 rows without exception

**Error message and/or stacktrace**
The query fails on the distributed_table_merged and runs successfully on the other 5 tables.
```
SELECT 1
FROM distributed_table_merged

Received exception from server (version 19.16.10):
Code: 171. DB::Exception: Received from clickhouse-server:9000. DB::Exception: Cannot convert column `1` because it is non constant in source stream but must be constant in result.
```

** Notes
It works fine in previous version `19.11.8`

**Comment by den-crane** [comment]
duplicate https://github.com/ClickHouse/ClickHouse/issues/8165

**Comment by azat** [comment]
Looks like it has been fixed (4a83e85893)?

```sql
CREATE TABLE merge AS system.one ENGINE = Merge(currentDatabase(), 'dist');
CREATE TABLE dist_1 AS system.one ENGINE = Distributed(test_shard_localhost, system, one);
CREATE TABLE dist_2 AS system.one ENGINE = Distributed(test_shard_localhost, system, one);
-- tried also test_cluster_two_shards_localhost
```

```
SELECT 1
FROM merge

┌─1─┐
│ 1 │
└───┘
┌─1─┐
│ 1 │
└───┘

2 rows in set. Elapsed: 0.033 sec. 
```

Or this schema is too simple?
